### ISSUE TITLE
hasReadWritePermission() incorrectly returns true

### ISSUE THREAD

**Issue by fabledfoe** [EddyVerbruggen/Calendar-PhoneGap-Plugin#342]
Testing on Android 7.1.1 and noticed the plugin crashing my app because of permissions.  Calling the hasReadWritePermission() function returns true, although the app currently does not have permissions granted (and had not previously prompted).  The manifest does automatically add the uses-permission lines at build time, so perhaps it is mistakenly reading those values as granted permissions?

Using the latest build 4.5.5

**Comment by EddyVerbruggen** [comment]
Whether or not `true` is returned als depends on the target SDK, on Android < 23 the app doesn't prompt for permission as the xml definition is sufficient.

**Comment by fabledfoe** [comment]
That is correct, although I am targeting 23 in this case.  If the SDK was lower, the app wouldn't crash.

**Comment by EddyVerbruggen** [comment]
OK - Can you show your `hasReadWritePermission` code?

**Comment by fabledfoe** [comment]
For debugging, I just copy/pasted your example which shows an alert, then commented out my call to the plugin.  The alert shows true.

```
window.plugins.calendar.hasReadWritePermission(
				function(result) {
					// if this is 'false' you probably want to call 'requestReadWritePermission' now
					alert(result);
				}
			);
```

**Comment by EddyVerbruggen** [comment]
I can't see what's wrong with [these lines](https://github.com/EddyVerbruggen/Calendar-PhoneGap-Plugin/blob/master/src/android/nl/xservices/plugins/Calendar.java#L147-L148) really.. have you tried removing and readding the app to clear the permissions? And can I perhaps get an APK to run it on my Android7 device?

**Comment by fabledfoe** [comment]
I agree - it looks very simple and I can't seem to figure out what would be different in this case.  I just booted an emulator with 6.0 and got the same result: if I leave in the plugin call - crashes due to permission error.  If I comment out and put in the check, it says true.  Just to see, I added the requestReadWritePermission call immediately after, but of course it doesn't fire due to the reported value.

I did also uninstall/reinstall the app with no change, but the fact that an emulator also fails means something else is up.

I wish I could give you the apk, but the app currently only runs on our internal servers while in development phase.  

I have these lines that appear automatically added to AndroidManifest.xml at build time:
    <uses-permission android:name="android.permission.READ_CALENDAR" />
    <uses-permission android:name="android.permission.WRITE_CALENDAR" />

The plugin has worked successfully before with this app - I have just been in the process of updating all plugins and came across this issue today while testing.  

Is there a minimum cordova android platform version required?  I just checked and currently have 4.1.1.

**Comment by EddyVerbruggen** [comment]
4.1.1 should be fine. If nothing else helps.. can you build with targetsdk 22 to circumvent this?

**Comment by fabledfoe** [comment]
Yeah, rebuild with target 22 avoids it for now.  If I can find any more pertinent info in my research, I'll post here.
